Qt6 build warns about: format specifies type 'int' but the argument has type 'qsizetype'

### What is the bug or the crash?

*as a developer* 
building Qt6 with GRASS8 warns about
`src/providers/grass/qgis.r.in.cpp:161:101: error: format specifies type 'int' but the argument has type 'qsizetype' (aka 'long long')`

https://github.com/qgis/QGIS/blob/14e27c2075fc78369d37649a34342e42771da3fb/src/providers/grass/qgis.r.in.cpp#L161

see https://cdash.orfeo-toolbox.org/viewBuildError.php?buildid=94783

### Steps to reproduce the issue

compile WITH_QT6=ON and WITH_GRASS8=ON same as in https://github.com/qgis/QGIS/pull/47887 - but set WERROR=false

### Versions

3.24.1 and master

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

not asking for support, just an issue tracker for developers

--- a/src/providers/grass/qgis.r.in.cpp
+++ b/src/providers/grass/qgis.r.in.cpp
@@ -158,7 +158,7 @@ int main( int argc, char **argv )
 
     if ( byteArray.size() != expectedSize )
     {
-      G_fatal_error( "Wrong byte array size, expected %d bytes, got %d, row %d / %d", expectedSize, byteArray.size(), row, rows );
+      G_fatal_error( "Wrong byte array size, expected %d bytes, got %lld, row %d / %d", expectedSize, static_cast<long long>( byteArray.size() ), row, rows );
       return 1;
     }
 

